src/django_components/component.py

```diff
@@ -103,14 +101,6 @@
 
 COMP_ONLY_FLAG = "only"
 
-# Define TypeVars for args and kwargs
-ArgsType = TypeVar("ArgsType", bound=tuple, contravariant=True, default=Any)
-KwargsType = TypeVar("KwargsType", bound=Mapping[str, Any], contravariant=True, default=Any)
-SlotsType = TypeVar("SlotsType", bound=Mapping[SlotName, SlotContent], default=Any)
-DataType = TypeVar("DataType", bound=Mapping[str, Any], covariant=True, default=Any)
-JsDataType = TypeVar("JsDataType", bound=Mapping[str, Any], default=Any)
-CssDataType = TypeVar("CssDataType", bound=Mapping[str, Any], default=Any)
-
 
 # NOTE: `ReferenceType` is NOT a generic pre-3.9
```

Comment:
While this is not part of this PR, I think this is solved by typing_extensions?

Reply:
Unfortunately not, the `ReferenceType` comes from the `weakref` module.